### PR TITLE
Using rich defined percentage type

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

- Adopt rich defined `Percentage` type for percentage representing field instead of primitive integer
- Update docs reflect changes.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Support string with suffix `%` passed as percentage representing fields' value beside primitive integer value
```

### CODE REVIEW

File: pkg/config/deployment_cloudrun.go [hunk 1]
```diff
@@ -47,5 +47,5 @@ type CloudRunSyncStageOptions struct {
 // CloudRunPromoteStageOptions contains all configurable values for a CLOUDRUN_PROMOTE stage.
 type CloudRunPromoteStageOptions struct {
 	// Percentage of traffic should be routed to the new version.
-	Percent int `json:"percent"`
+	Percent Percentage `json:"percent"`
```
Comment (nghialv) [inline pkg/config/deployment_cloudrun.go:50]: Oops, the field name and tag should be `percentage`.

Comment (khanhtc1202) [reply pkg/config/deployment_cloudrun.go:50]: Both `Percent` and `Percentage` is lgtm but change to `Percentage` for now is a breaking change so I suggest not doing that 😄

Comment (nghialv) [reply pkg/config/deployment_cloudrun.go:50]: Yes, I know. We should not change it. I just wanted to say that I made a mistake on them. Percent is often used to express from 0 to 1, but percentage is from 0 to 100.

Comment (khanhtc1202) [reply pkg/config/deployment_cloudrun.go:50]: I see, that diff is a dangerous thing 😱

**Comment by khanhtc1202** [comment]
@nakabonne PTAL 🙏

**Review by nakabonne — APPROVED** [review]
👍

**Pull request merged**